source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs

```diff
@@ -85,80 +86,70 @@ private Charge()
-
-        public void StopCharge(Instant endDate)
-        {
-            throw new NotImplementedException();
-        }
-
         /// <summary>
         /// Use this method to update the charge periods timeline of a charge upon receiving a charge update request
         /// Please see the persist charge documentation where the update flow is covered:
         /// https://github.com/Energinet-DataHub/geh-charges/tree/main/docs/process-flows#persist-charge
         /// </summary>
         /// <param name="newChargePeriod">New Charge Period from update charge request</param>
         /// <exception cref="ArgumentNullException">Throws when <paramref name="newChargePeriod"/> is empty</exception>
-        public void UpdateCharge(ChargePeriod newChargePeriod)
+        public void Update(ChargePeriod newChargePeriod)
         {
             if (newChargePeriod == null) throw new ArgumentNullException(nameof(newChargePeriod));
 
-            RemoveAllPeriodsFromNewPeriodStart(newChargePeriod);
-            HandleAnyOverlappingPeriod(newChargePeriod);
+            if (newChargePeriod.EndDateTime != InstantExtensions.GetEndDefault())
+            {
+                throw new InvalidOperationException("Charge update must not have bound end date.");
+            }
+
+            if (_periods.Exists(p => p.StartDateTime < newChargePeriod.StartDateTime))
+            {
+                StopExistingPeriod(newChargePeriod.StartDateTime);
+            }
+
+            RemoveAllSubsequentPeriods(newChargePeriod.StartDateTime);
             _periods.Add(newChargePeriod);
-            Validate();
         }
 
-        private void HandleAnyOverlappingPeriod(ChargePeriod newChargePeriod)
+        /// <summary>
+        /// Use this method to stop a charge upon receiving a stop charge request
+        /// </summary>
+        /// <param name="stopDate"></param>
+        /// <exception cref="InvalidOperationException">Throws when no charge periods found on charge.</exception>
+        public void Stop(Instant? stopDate)
         {
-            var previousPeriod = _periods
-                .SingleOrDefault(p =>
-                p.EndDateTime > newChargePeriod.StartDateTime &&
-                p.StartDateTime < newChargePeriod.StartDateTime);
-
-            if (previousPeriod != null)
+            if (stopDate == null)
             {
-                var newPreviousPeriod = new ChargePeriod(
-                    previousPeriod.Id,
-                    previousPeriod.Name,
-                    previousPeriod.Description,
-                    previousPeriod.VatClassification,
-                    previousPeriod.TransparentInvoicing,
-                    previousPeriod.StartDateTime,
-                    newChargePeriod.StartDateTime);
-                _periods.Remove(previousPeriod);
-                _periods.Add(newPreviousPeriod);
+                throw new InvalidOperationException("Could not stop charge. EndDate not set.");
```

Comment:
```suggestion
                throw new ArgumentNullException("Could not stop charge. EndDate not set.");
```
and update description in line 118